### DISCUSSION TITLE
Why I do need this?

### DISCUSSION THREAD

**Discussion by jbovone** [react-navigation/react-navigation#10549] [General]
hi guys!
I'm relatively new in react native. And I was checking up the most used Api's for native development to see what goodies are available to me for this journey.

`const App = () => {
  const [dest, setDest] = useState('home');
  switch (dest) {
    case 'home': return <Home navigate={setDest}/>
    case 'contact': return <Contact navigate={setDest}/>
    default: return <Woops/>   
  }
};`

I have check the docs but instead of filling me with howto's.. I believe that this is probably first thing in the hierarchy to figure out. 

Why this Api is so extensive, why beats that simple minded routing of above? 
What are you fighting here guys, teach me =P? 

Thank you!

**Comment by satya164** [comment]
React Navigation implements the navigation patterns found in native apps so your app will look and feel like other mobile apps. There are a lot of things such as gestures, animations, UI elements matching native interface, nested navigation, deep linking, and many more.

The way you wrote the code will functionally work but isn't going to feel like a proper mobile app, and will need a lot of custom code to implement the functionality which is built into the library.
